### PR TITLE
add oms_exportSnapshot API

### PR DESCRIPTION
**Pull request by arun3688**
### Purpose

This PR adds new API `oms_exportSnapshot` which adds layering to xml file for identifyfing different kind of ssp files like
 
1. .ssd
2.  .ssv
3.  .ssm

### Approach
We use a new tagging 
```
<oms:snapshot>   
<oms:ssd_file>      
<oms:ssv_file>     
<oms:ssm_file>   
```

### Example

```
<?xml version="1.0"?>
<oms:snapshot>
 <oms:ssd_file name="SystemStructure.ssd">
   <ssd:SystemStructureDescription xmlns:ssc="http://ssp-standard.org/SSP1/SystemStructureCommon" xmlns:ssd="http://ssp-standard.org/SSP1/SystemStructureDescription" xmlns:ssv="http://ssp-standard.org/SSP1/SystemStructureParameterValues" xmlns:ssm="http://ssp-standard.org/SSP1/SystemStructureParameterMapping" xmlns:ssb="http://ssp-standard.org/SSP1/SystemStructureSignalDictionary" xmlns:oms="https://raw.githubusercontent.com/OpenModelica/OMSimulator/master/schema/oms.xsd" name="exportSnapshot" version="1.0">
     <ssd:System name="root">
       <ssd:Annotations>
         <ssc:Annotation type="org.openmodelica">
           <oms:SimulationInformation>
             <oms:FixedStepMaster description="oms-ma" stepSize="0.100000" absoluteTolerance="0.000100" relativeTolerance="0.000100" />
           </oms:SimulationInformation>
         </ssc:Annotation>
       </ssd:Annotations>
       <ssd:Connectors>
         <ssd:Connector name="C1" kind="input">
           <ssc:Real />
         </ssd:Connector>
       </ssd:Connectors>
       <ssd:ParameterBindings>
         <ssd:ParameterBinding source="resources/exportSnapshot.ssv" />
       </ssd:ParameterBindings>
       <ssd:Elements>
         <ssd:Component name="add" type="application/x-fmu-sharedlibrary" source="resources/0001_add.fmu">
           <ssd:Connectors>
             <ssd:Connector name="u1" kind="input">
               <ssc:Real />
               <ssd:ConnectorGeometry x="0.000000" y="0.333333" />
             </ssd:Connector>
             <ssd:Connector name="u2" kind="input">
               <ssc:Real />
               <ssd:ConnectorGeometry x="0.000000" y="0.666667" />
             </ssd:Connector>
             <ssd:Connector name="y" kind="output">
               <ssc:Real />
               <ssd:ConnectorGeometry x="1.000000" y="0.500000" />
             </ssd:Connector>
             <ssd:Connector name="k1" kind="parameter">
               <ssc:Real />
             </ssd:Connector>
             <ssd:Connector name="k2" kind="parameter">
               <ssc:Real />
             </ssd:Connector>
           </ssd:Connectors>
         </ssd:Component>
       </ssd:Elements>
       <ssd:Connections />
     </ssd:System>
     <ssd:DefaultExperiment startTime="0.000000" stopTime="1.000000">
       <ssd:Annotations>
         <ssc:Annotation type="org.openmodelica">
           <oms:SimulationInformation resultFile="exportSnapshot_res.mat" loggingInterval="0.000000" bufferSize="10" signalFilter="" />
         </ssc:Annotation>
       </ssd:Annotations>
     </ssd:DefaultExperiment>
   </ssd:SystemStructureDescription>
 </oms:ssd_file>
 <oms:ssv_file name="resources/exportSnapshot.ssv">
   <ssv:ParameterSet xmlns:ssc="http://ssp-standard.org/SSP1/SystemStructureCommon" xmlns:ssv="http://ssp-standard.org/SSP1/SystemStructureParameterValues" version="1.0" name="parameters">
     <ssv:Parameters>
       <ssv:Parameter name="C1">
         <ssv:Real value="-10" />
       </ssv:Parameter>
       <ssv:Parameter name="add.u1">
         <ssv:Real value="10" />
       </ssv:Parameter>
       <ssv:Parameter name="add.k1">
         <ssv:Real value="30" />
       </ssv:Parameter>
     </ssv:Parameters>
   </ssv:ParameterSet>
 </oms:ssv_file>
</oms:snapshot>
```

### CODE REVIEW

**Comment by arun3688** [comment]
@adeas31 , now we will use `oms_exportSnapshot()` to get all the xml files contents, instead of `oms_list()`  and then we will use 
`oms_importSnapshot()` (work in progress) instead of 'oms_loadSnapshot()'

**Comment by lochel** [comment]
Results are looking good. Will read the changes later today or tomorrow.

File: src/OMSimulatorLib/System.cpp [hunk 1]
```diff
@@ -503,16 +503,16 @@ oms_status_enu_t oms::System::importFromSSD(const pugi::xml_node& node, const st
...
         if (result) // check from ssv file
         {
           parameterSet = ssvdoc.document_element(); // ssv:ParameterSet
           parameters = parameterSet.child(oms::ssp::Version1_0::ssv::parameters);
         }
-        else if((ssvNode = getModel()->getSnapshot().child(oms::ssp::Version1_0::ssv_file))) // check in memory oms:ssv_file
+        else if (getModel()->getSnapshot().child(oms::ssp::Version1_0::ssv_file)) // check in memory oms:ssv_file
         {
-          parameterSet = ssvNode.child(oms::ssp::Version1_0::ssv::parameter_set); // ssv:ParameterSet
+          parameterSet = getModel()->getSnapshot().child(oms::ssp::Version1_0::ssv_file).child(oms::ssp::Version1_0::ssv::parameter_set); // ssv:ParameterSet
```
Comment (lochel) [inline src/OMSimulatorLib/System.cpp:515]: Did you test it with a model that contains several ssv files?

Comment (arun3688) [reply src/OMSimulatorLib/System.cpp]: This is fixed with  https://github.com/OpenModelica/OMSimulator/pull/820/commits/76203f76451f5a567cdc404e25d1443042481f00

File: doc/UsersGuide/source/api/exportSnapshot.rst [hunk 2]
```diff
@@ -0,0 +1,41 @@
+#CAPTION#
+list
```
Comment (lochel) [inline doc/UsersGuide/source/api/exportSnapshot.rst:2]: exportSnapshot

**Review by lochel — CHANGES_REQUESTED** [review]

File: doc/UsersGuide/source/api/exportSnapshot.rst (hunk 2 above)
Comment (arun3688) [reply doc/UsersGuide/source/api/exportSnapshot.rst]: done

**Review by lochel — APPROVED** [review]

**Pull request merged**